src/osp/Active/SysAreaAssociate.cpp

```diff
@@ -39,60 +39,39 @@ using osp::universe::UCompActiveArea;
 
 using osp::universe::Vector3g;
 
-void SysAreaAssociate::update_scan(ActiveScene& rScene)
+void SysAreaAssociate::update_consume(ActiveScene& rScene)
 {
-    ACompAreaLink *pArea = try_get_area_link(rScene);
+    ACompAreaLink *pLink = try_get_area_link(rScene);
 
-    if (pArea == nullptr)
+    if (pLink == nullptr)
     {
         return;
     }
 
-    Universe &rUni = pArea->m_rUniverse;
-    Satellite areaSat = pArea->m_areaSat;
-    auto &areaUComp = rUni.get_reg().get<UCompActiveArea>(areaSat);
-
-    // Clear enter/leave event vectors
-
-    pArea->m_enter.clear();
-    pArea->m_leave.clear();
+    Universe &rUni = pLink->m_rUniverse;
+    Satellite areaSat = pLink->m_areaSat;
+    auto &rAreaUComp = rUni.get_reg().get<UCompActiveArea>(areaSat);
 
-    // Deal with activating satellites that have a UCompActivationRadius
-    // Satellites that have an Activation Radius have a sphere around them. If
-    // this sphere overlaps the ActiveArea's sphere 'm_areaRadius', then it
-    // will be activated.
+    pLink->m_enter = std::exchange(rAreaUComp.m_enter, {});
+    pLink->m_leave = std::exchange(rAreaUComp.m_leave, {});
 
-    auto viewActRadius = rUni.get_reg().view<
-            UCompActivationRadius, UCompActivatable>();
+    Vector3g deltaTotal{};
 
-    for (Satellite sat : viewActRadius)
+    for (Vector3g const delta : std::exchange(rAreaUComp.m_moved, {}))
```

Comment:
Vector3g has s no move semantics right? If so, const& would be slightly better on low optimization levels.